### PR TITLE
Refs #24121 -- Added __repr__() to URLNode.

### PR DESCRIPTION
**Pull request by rohithpr**
[Ticket](https://code.djangoproject.com/ticket/24121)

There are a number of changes that can be made as a part of the ticker linked above, this PR addresses only one of them and does not close the ticket.

### CODE REVIEW

File: django/template/defaulttags.py [hunk 1]
```diff
@@ -406,6 +406,12 @@ def __init__(self, view_name, args, kwargs, asvar):
         self.kwargs = kwargs
         self.asvar = asvar
 
+    def __repr__(self):
+        return (
+            f'<{self.__class__.__name__} view_name={self.view_name} '
```
Comment (felixxm) [inline django/template/defaulttags.py:411]: `self.__class__.__qualname__`

File: django/template/defaulttags.py [hunk 2]
```diff
@@ -406,6 +406,12 @@ def __init__(self, view_name, args, kwargs, asvar):
...
+            f'args={self.args} kwargs={self.kwargs} asvar={self.asvar}>'
```
Comment (felixxm) [inline django/template/defaulttags.py:412]: I would wrap `asvar` and `view_name` in quotes and use `%`-formatting, e.g.:
```python
    def __repr__(self):
        return "<%s view_name='%s' args=%s kwargs=%s as='%s'>" % (
            self.__class__.__name__,
            self.view_name,
            repr(self.args),
            repr(self.kwargs),
            self.asvar,
        )
```

Comment (felixxm) [inline django/template/defaulttags.py:412]: `asvar` -> `as`

File: tests/template_tests/test_url_node.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        url_node = URLNode(
+            view_name='named-view',
+            args=(123, 456),
+            kwargs={'foo': 'bar'},
+            asvar='as-view'
+        )
+        self.assertEqual(
+            repr(url_node),
+            "<URLNode view_name=named-view args=(123, 456) kwargs={'foo': 'bar'} asvar=as-view>"
+        )
```
Comment (felixxm) [inline tests/template_tests/test_url_node.py:17]: We can move this class to the `tests/template_tests/syntax_tests/test_url.py`.

**Review by felixxm — COMMENTED** [review]
@rohithpr Thanks for this patch :+1:

File: django/template/defaulttags.py [hunk 4]
```diff
@@ -406,6 +406,15 @@ def __init__(self, view_name, args, kwargs, asvar):
         self.kwargs = kwargs
         self.asvar = asvar
 
+    def __repr__(self):
+        return "<%s view_name='%s' args=%s kwargs=%s as='%s'>" % (
+            self.__class__.__qualname__,
+            self.view_name,
+            repr(self.args),
+            repr(self.kwargs),
+            self.asvar
```
Comment (felixxm) [inline django/template/defaulttags.py]: `asvar` can be `None`, so I added `repr()`.

**Review by felixxm — COMMENTED** [review]
@rohithpr Thanks for updates :+1: Welcome aboard :boat: 

I pushed minor edits.

**Comment by rohithpr** [comment]
Thank you!

**Pull request merged**